pages/posts/the-architectural-scope-of-json-schema.md

```diff
@@ -0,0 +1,76 @@
+
+An application may only provide an interface over data persistence (e.g. a Postgres web service), meaning that there's no need for any business logic.  Conversely, another application may provide a computational service (e.g. image processing) where there's no need to persist data.
+
+For the recent conversations I've had, this second scenario seems to be the case: an application is created that just evaluates instances against schemas.  But this doesn't mean that the application and implementation *are* the same thing.
+
+Within these applications the implementation, which for these applications _is_ the business logic, is still a separate component from the interface.  And it's important to recognize that JSON Schema as a specification can only cover the implementation part.
+
+## Why any of this matters
+
+It all comes back to what I touched on this in the opening section: JSON Schema needs to define inputs and outputs.  This comprises a minimal API that implementations and applications can use to communicate with each other.
+
+When the line between implementation and application is blurred, it's natural to think that the specification is imposing requirements on how the application communicates with its users.  But that's not the case.
+
+It is impossible for JSON Schema to know the needs of an application's users, and so it's impractical for the specification to attempt to define input and output requirements to which applications must adhere.
+
+Users of different applications have different needs.  Even when you consider two applications that essentially just provide a UX for implementations, say a web app and a CLI, the UX needs of their users are vastly different, despite the two applications doing basically the same thing.
+
+## The scope of the specification
+
+As a result of everything discussed above, it follows that the specification's input and output requirements are only applicable when there is a clear communication seam between an application and a JSON Schema implementation.
+
+The specification recognizes that programming languages and frameworks likely will not be dealing with textual JSON, but rather they will use data models defined within the limitations of that language.  As such, it defines input and output in terms of abstract JSON data and JSON Schema models so that implementations are free to use what they have at their disposal.
+
+Specifically, these requirements only pertain to standalone implementations that are provided as general-use representations of the JSON Schema specification to be consumed by unknown parties.
+
+Applications which have integrated implemenations or application/implemenatation pairs which have specialized contracts need not adhere to these requirements because these arrangements are out of scope of the specification.
```

Comment:
typo: implemenations -> implemenTations

Also, is this intending to create "integrated implementations" as a new category with different requirements from other implementations?  What would the implications be for claiming conformance to JSON Schema if there is no JSON Schema interface that can be hooked up to a test suite?

Reply:
I do that. :)